chains/evm/contracts/onRamp/ExecutorOnRamp.sol

```diff
@@ -0,0 +1,203 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+import {IExecutorOnRamp} from "../interfaces/IExecutorOnRamp.sol";
+
+import {Client} from "../libraries/Client.sol";
+import {Ownable2StepMsgSender} from "@chainlink/contracts/src/v0.8/shared/access/Ownable2StepMsgSender.sol";
+
+import {EnumerableSet} from
+  "@chainlink/contracts/src/v0.8/vendor/openzeppelin-solidity/v5.0.2/contracts/utils/structs/EnumerableSet.sol";
+
+/// @notice The ExecutorOnRamp configures the supported destination chains and CCV limits for an executor.
+contract ExecutorOnRamp is Ownable2StepMsgSender, IExecutorOnRamp {
+  using EnumerableSet for EnumerableSet.AddressSet;
+  using EnumerableSet for EnumerableSet.UintSet;
+
+  error ExceedsMaxCCVs(uint256 provided, uint256 max);
+  error InvalidCCV(address ccv);
+  error InvalidDestChain(uint64 destChainSelector);
+  error InvalidMaxPossibleCCVsPerMsg(uint256 maxPossibleCCVsPerMsg);
+
+  event AllowlistUpdated(bool enabled);
+  event CCVAdded(address indexed ccv);
+  event CCVRemoved(address indexed ccv);
+  event ConfigSet(DynamicConfig dynamicConfig);
+  event DestChainAdded(uint64 indexed destChainSelector);
+  event DestChainRemoved(uint64 indexed destChainSelector);
+
+  /// @dev Struct that defines the dynamic configuration.
+  // solhint-disable-next-line gas-struct-packing
+  struct DynamicConfig {
+    // Max(required ccvs + optional ccvs).
+    // Limits the number of ccvs that the executor needs to search for results from.
+    uint8 maxCCVsPerMsg;
+  }
+
+  /// @notice Whether or not the CCV allowlist is enabled.
+  bool public s_allowlistEnabled;
+  /// @notice The set of CCVs that the executor supports.
+  /// @dev Addresses correspond to the CCVOnRamp for each CCV.
+  EnumerableSet.AddressSet private s_allowedCCVs;
+  /// @notice The set of destination chains that the executor supports.
+  EnumerableSet.UintSet private s_allowedDestChains;
+  /// @dev The dynamic config.
+  DynamicConfig private s_dynamicConfig;
+
+  string public constant typeAndVersion = "ExecutorOnRamp 1.7.0-dev";
+
+  constructor(
+    DynamicConfig memory dynamicConfig
+  ) {
+    _setDynamicConfig(dynamicConfig);
+  }
+
+  // ================================================================
+  // │                           Config                             │
+  // ================================================================
+
+  /// @notice Returns the dynamic config.
+  /// @return dynamicConfig The config.
+  function getDynamicConfig() external view returns (DynamicConfig memory dynamicConfig) {
```

Comment:
Not sure if it's worth to have a dynamic config if there's only a single entry. Is there any tooling benefit from having the same call (low context overhead for executing proposals?) or should we just have this single value be a top level arg. Having it as top level means we can pack it with s_allowlistEnabled which saves gas.

Reply:
You could still have the functions to set dynamic but rm the struct and just have it take a uint

Reply:
No significant tooling benefit, I can pack it

Reply:
https://github.com/smartcontractkit/chainlink-ccip/pull/1157/commits/d0283cb3c15560bd1ea7a82638c2edc05295bd05